aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -380,11 +384,27 @@ module aptos_framework::account {
     }
 
     /// A resource account is used to manage resources independent of an account managed by a user.
-    public fun create_resource_account(source: &signer, seed: vector<u8>): (signer, SignerCapability) {
-        let addr = create_resource_address(&signer::address_of(source), seed);
-        let signer = create_account_unchecked(copy addr);
-        let signer_cap = SignerCapability { account: addr };
-        (signer, signer_cap)
+    public fun create_resource_account(source: &signer, seed: vector<u8>): (signer, SignerCapability) acquires Account {
+        let resource_addr = create_resource_address(&signer::address_of(source), seed);
+        let resource = if (exists_at(resource_addr)) {
+            let account = borrow_global<Account>(resource_addr);
+            assert!(
+                option::is_none(&account.signer_capability_offer.for),
+                error::already_exists(ERESOURCE_ACCCOUNT_EXISTS),
+            );
+            assert!(
+                account.sequence_number == 0,
+                error::invalid_state(EACCOUNT_ALREADY_USED),
+            );
+            create_signer(resource_addr)
+        } else {
+            let resource = create_account_unchecked(resource_addr);
+            let account = borrow_global_mut<Account>(resource_addr);
+            account.signer_capability_offer.for = option::some(resource_addr);
```

Comment:
So what is this meant to achieve?

Reply:
I believe this is meant to ensure that `SignerCapability` can only be offered once? 

It seems like you should move this assignment out of the else branch to set the capability, otherwise you can repeatedly create `SignerCapability`s?

Reply:
ahh, yes, yes... good call.